site/source/docs/tools_reference/settings_reference.rst

```diff
@@ -0,0 +1,2376 @@
+- ASYNCIFY_LAZY_LOAD_CODE
+- WASM2JS (WASM=0)
+
+SPLIT_MODULE
+============
+
+Generate code to loading split wasm modules.
+This option will automatically generate two wasm files as output, one
+with the ``.orig`` suffix and one without.  The default file (without
+the suffix) when run will generate instrumentation data can later be
+fed into wasm-split (the binaryen tool).
+As well as this the generated JS code will contains help functions
+to loading split modules.
+
+AUTOLOAD_DYLIBS
+===============
+
+For MAIN_MODULE builds, automatically load any dynamic library dependencies
+on startup, before loading the main module.
+
+ALLOW_UNIMPLEMENTED_SYSCALLS
+============================
+
+Include unimplemented JS syscalls to be included in the final output.  This
+allows programs that depend on these syscalls at runtime to be compiled, even
+though these syscalls will fail (or do nothing) at runtime.
+
+TRUSTED_TYPES
+=============
+
+Allow calls to Worker(...) and importScripts(...) to be Trusted Types compatible.
+Trusted Types is a Web Platform feature designed to mitigate DOM XSS by restricting
+the usage of DOM sink APIs. See https://w3c.github.io/webappsec-trusted-types/.
+
+POLYFILL
+========
+
+When targeting older browsers emscripten will sometimes require that
+polyfills be included in the output.  If you would prefer to take care of
+polyfilling yourself via some other mechanism you can prevent emscripten
+from generating these by passing ``-sNO_POLYFILL`` or ``-sPOLYFILL=0``
+With default browser targets emscripten does not need any polyfills so this
+settings is *only* needed when also explicitly targeting older browsers.
+
+RUNTIME_DEBUG
+=============
+
+If true, add tracing to core runtime functions.
+This setting is enabled by default if any of the following debugging settings
+are enabled:
+- PTHREADS_DEBUG
+- DYLINK_DEBUG
+- LIBRARY_DEBUG
+- GL_DEBUG
+- OPENAL_DEBUG
+- EXCEPTION_DEBUG
+- SYSCALL_DEBUG
+- WEBSOCKET_DEBUG
+- SOCKET_DEBUG
+- FETCH_DEBUG
+
+LEGACY_RUNTIME
+==============
+
+Include JS library symbols that were previously part of the default runtime.
+Without this, such symbols can be made available by adding them to
+DEFAULT_LIBRARY_FUNCS_TO_INCLUDE, or via the dependencies of another JS
+library symbol.
+
+SIGNATURE_CONVERSIONS
+=====================
+
+User-defined functions to wrap with signature conversion, which take or return
+pointer argument. Only affects MEMORY64=1 builds, see create_pointer_conversion_wrappers
+in emscripten.py for details.
+Use _ for non-pointer arguments, p for pointer/i53 arguments, and P for optional pointer/i53 values.
+Example use -sSIGNATURE_CONVERSIONS=someFunction:_p,anotherFunction:p
+
+OFFSCREEN_FRAMEBUFFER_FORBID_VAO_PATH
+=====================================
+
+Internal (testing only): Disables the blitOffscreenFramebuffer VAO path.
+
+TEST_MEMORY_GROWTH_FAILS
+========================
+
+Internal (testing only): Forces memory growing to fail.
+
+LEGACY_SETTINGS
```

Comment:
This line and onwards seems unnecessary?

Reply:
Done